src/Halogen/HTML/Events.purs

```diff
@@ -211,25 +206,27 @@ clipboardHandler = unsafeCoerce
 
 -- | Attaches event handler to event `key` with getting `prop` field as an
 -- | argument of `handler`.
-addForeignPropHandler :: forall r i value. IsForeign value => EventType -> String -> (value -> Maybe i) -> IProp r i
-addForeignPropHandler key prop f =
-  handler key (either (const Nothing) f <<< runExcept <<< readProp prop <<< toForeign <<< EE.currentTarget)
+addForeignPropHandler :: forall r i value. EventType -> String -> (Foreign -> F value) -> (value -> Maybe i) -> IProp r i
+addForeignPropHandler key prop reader f =
+  handler key (either (const Nothing) (maybeRead >=> f) <<< runExcept <<< readProp prop <<< toForeign <<< EE.currentTarget)
+  where
+    maybeRead = reader >>> runExcept >>> either (const Nothing) Just
```

Comment:
I think it's fine to just apply `readString`, etc. where they're used - we only have these 4 cases of it anyway. I don't mind whether the `read` part of it is positioned at the start or the end either, I just happened to write it that way when checking the types work out for my version!